nomad/jobs_endpoint.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package nomad
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/go-set/v2"
+	"github.com/hashicorp/nomad/acl"
+	"github.com/hashicorp/nomad/nomad/state"
+	"github.com/hashicorp/nomad/nomad/state/paginator"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+func NewJobsEndpoint(s *Server, ctx *RPCContext) *Jobs {
+	return &Jobs{
+		srv:    s,
+		ctx:    ctx,
+		logger: s.logger.Named("jobs"),
+	}
+}
+
+type Jobs struct {
+	srv    *Server
+	ctx    *RPCContext
+	logger hclog.Logger
+}
```

Comment:
It's not clear to me that this should be a different RPC endpoint struct than the existing `Job` struct. Having it in its own endpoint means it gets its own metrics collection (which is a little weird) and we have to create the struct, it's logger, etc. for every RPC request.

If we're just wary of adding more methods to the big `nomad/jobs_endpoint.go` file, we can always split it out into a separate file.

Reply:
I started it like this to have a clean slate for my mental model.  the only argument I have to keep it separate is that it's unique in accepting a list of jobs as request _inputs_, but it's a pretty weak point, and plenty of other "Job" endpoints deal with plural jobs as responses.

all that to say, I'll scoot it over to `Job`!

Reply:
done in [1f593d9](https://github.com/hashicorp/nomad/pull/20130/commits/1f593d937dc49e887ab73311af9c31084c6a3823)